mathesar/api/ui/permissions/ui_records.py

```diff
@@ -0,0 +1,65 @@
+from rest_access_policy import AccessPolicy
+
+from mathesar.api.utils import get_table_or_404
+from mathesar.models.users import DatabaseRole, Role, SchemaRole
+
+
+class RecordAccessPolicy(AccessPolicy):
+    """
+    Anyone can view a Record as long as they are a superuser or have
+    at least a Viewer access to the Schema or the Database of the Table.
+    The permissions trickle down, so if someone has a Viewer Access for a Database
+    They automatically become a Schema Viewer
+    Refer https://wiki.mathesar.org/en/product/specs/users-permissions#database-permissions
+    Only superuser or schema/database Manager/Editor can delete/modify/create a Record
+    """
+    statements = [
+        {
+            'action': ['list', 'retrieve'],
+            'principal': '*',
+            'effect': 'allow',
+            'condition_expression': ['(is_superuser or is_table_viewer)']
+        },
+        {
+            'action': ['update', 'partial_update', 'create', 'delete'],
```

Comment:
Why was `destroy` inappropriate here? That would allow reusing the already-existing access policy, no? Can you please explain the need for this separate access policy with this single change?

Reply:
I did that to seperate ui permissions from db permissions, we can use existing use db permissions too.

Reply:
Please just use the existing permissions if they work. We need to reorganize this section of the code at some point, and I'd prefer to keep it simpler for now.

Reply:
then we'll need to add delete there since action name is delete along with destroy,

Reply:
Yes that's fine @aagman945